M2/Macaulay2/m2/newring.m2

```diff
@@ -11,6 +11,21 @@ needs "matrix2.m2" -- for lift
 -- new polynomial ring or quotient ring from old --
 ----------------------------------
 
+-- automate promotion
+promoteFromMap = method()
```

Comment:
Maybe call this `setupPromotionMethods` or something like that?

Reply:
And perhaps `commonEngineRingInitializations` should call it, similar to the `lift` methods?

Reply:
how about just `setupPromote`? so it's not too long

Reply:
Up to you. My plan is to eventually allow the notation `map(R, S) = f`.